crates/proof-of-sql/src/base/scalar/scalar_ext.rs

```diff
@@ -30,6 +31,26 @@ pub trait ScalarExt: Scalar {
     fn into_u256_wrapping(self) -> U256 {
         U256::from(Into::<[u64; 4]>::into(self))
     }
+
+    /// Converts a byte slice to a Scalar using a hash function, preventing collisions.
+    /// WARNING: Only up to 31 bytes (2^248 bits) are supported by `PoSQL` cryptographic
+    /// objects. This function masks off the last byte of the hash to ensure the result
+    /// fits in this range.
+    #[must_use]
+    fn from_byte_slice_via_hash(bytes: &[u8]) -> Self {
+        if bytes.is_empty() {
+            return Self::zero();
+        }
+
+        let mut hasher = tiny_keccak::Keccak::v256();
+        hasher.update(bytes);
+        let mut hashed_bytes = [0u8; 32];
+        hasher.finalize(&mut hashed_bytes);
+        let hashed_val =
+            U256::from_le_slice(&hashed_bytes).expect("32 bytes => guaranteed to parse as U256");
```

Comment:
NIT: we can avoid the expect (and the presumably slower checking required in `from_le_slice`)
```suggestion
        let mut hashed_bytes = [0u64; 4];
        hasher.finalize(hashed_bytes.as_mut_bytes());
        let hashed_val = U256::from(hashed_bytes );
```